components/layout_2020/positioned.rs

```diff
@@ -473,11 +480,16 @@ impl HoistedAbsolutelyPositionedBox {
         } = layout_style.content_box_sizes_and_padding_border_margin(&containing_block.into());
         let containing_block = &containing_block.into();
         let is_table = layout_style.is_table();
-
         let shared_fragment = self.fragment.borrow();
-        let static_position_rect = shared_fragment
+
+        // The static position rect was calculated assuming that the containing block would be
+        // established by the content box of some ancestor, but the actual containing block is
+        // established by the padding box. So we need to add the padding of that ancestor.
+        let mut static_position_rect = shared_fragment
             .static_position_rect
-            .to_logical(containing_block);
+            .outer_rect(-containing_block_padding);
+        static_position_rect.size = static_position_rect.size.max(Default::default());
```

Comment:
Maybe `Zero::zero()` or `LogicalSize::zero()` might be a bit clearer here?

Reply:
Used `PhysicalSize::zero()`.